### PR TITLE
PRO-1849 - Paymaster Updates

### PR DESCRIPTION
**Pull request by vignesha22**
## Description

- Removed paymaster initialisation from sdk init place and added to estimate step to specify how each userOp is submitted rather than global paymaster initialisation

## How Has This Been Tested?

- Tested locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Review by lbw33 — APPROVED** [review]

File: src/sdk/sdk.ts [hunk 1]
```diff
@@ -136,11 +130,16 @@ export class PrimeSdk {
...
-  async estimate(gasDetails?: TransactionGasInfoForUserOp) {
+  async estimate(paymasterDetails?: PaymasterApi, gasDetails?: TransactionGasInfoForUserOp) {
     if (this.userOpsBatch.to.length < 1) {
       throw new Error("cannot sign empty transaction batch");
     }
 
+    if (paymasterDetails?.url) {
+      const paymasterAPI = new VerifyingPaymasterAPI(paymasterDetails.url, Networks[this.chainId].contracts.entryPoint, paymasterDetails.context ?? {}, paymasterDetails.api_key, this.chainId)
+      this.etherspotWallet.setPaymasterApi(paymasterAPI)
+    } else this.etherspotWallet.setPaymasterApi(null);
```
Comment (ch4r10t33r) [inline src/sdk/sdk.ts:141]: Did this pass lint?

File: src/sdk/sdk.ts [hunk 2]
```diff
@@ -136,11 +130,16 @@ export class PrimeSdk {
...
   }
 
-  async estimate(gasDetails?: TransactionGasInfoForUserOp) {
+  async estimate(paymasterDetails?: PaymasterApi, gasDetails?: TransactionGasInfoForUserOp) {
     if (this.userOpsBatch.to.length < 1) {
       throw new Error("cannot sign empty transaction batch");
     }
 
+    if (paymasterDetails?.url) {
+      const paymasterAPI = new VerifyingPaymasterAPI(paymasterDetails.url, Networks[this.chainId].contracts.entryPoint, paymasterDetails.context ?? {}, paymasterDetails.api_key, this.chainId)
```
Comment (ch4r10t33r) [inline src/sdk/sdk.ts:139]: Does this code work with both stackup and pimlico paymasters?

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,4 +1,7 @@
 # Changelog
+## [1.2.9] - 2023-09-20
+### Breaking Changes
```
Comment (ch4r10t33r) [inline CHANGELOG.md:3]: Also add information on whether are recommending users to upgrade the SDK or this is just information/suggestion.

**Review by ch4r10t33r — CHANGES_REQUESTED** [review]

File: src/sdk/sdk.ts (hunk 1 above)
Comment (vignesha22) [reply src/sdk/sdk.ts:141]: yes

File: CHANGELOG.md (hunk 3 above)
Comment (vignesha22) [reply CHANGELOG.md:3]: okay

File: src/sdk/sdk.ts (hunk 2 above)
Comment (vignesha22) [reply src/sdk/sdk.ts:139]: As discussed, this will be tested and make the necessary changes in a separate ticket

**Comment by ch4r10t33r** [comment]
@vignesha22 pls resolve conflicts.

**Pull request merged**